core/src/main/java/org/mule/runtime/core/api/util/ClassUtils.java

```diff
@@ -372,36 +379,96 @@ public static Class loadClass(final String className, final ClassLoader classLoa
         }
       }
     }
-
     throw new NoSuchFieldException(String.format("Could not find field '%s' in class %s", fieldName,
-                                                 target.getClass().getName()));
+                                                 targetClass.getName()));
   }
 
-  public static void setFieldValue(Object target, String fieldName, Object value, boolean recursive)
+  /**
+   * Gets a field value of a given object
+   * @param target object to get the field value from
+   * @param fieldName then name of the field
+   * @param recursive flag to lookup in subclasses
+   * @param <T> The expected type of the field.
+   * @return the value of the field for the target object
+   * @throws IllegalAccessException when the field is not reachable
+   * @throws NoSuchFieldException when the field does not exists
+   */
+  public static <T> T getFieldValue(Object target, String fieldName, boolean recursive)
       throws IllegalAccessException, NoSuchFieldException {
-    Class<?> clazz = target.getClass();
-    Field field;
-    while (!Object.class.equals(clazz)) {
-      try {
-        field = clazz.getDeclaredField(fieldName);
-        field.setAccessible(true);
-        field.set(target, value);
+    Field f = getField(target.getClass(), fieldName, recursive);
+    f.setAccessible(true);
+    return (T) f.get(target);
+  }
 
-        return;
-      } catch (NoSuchFieldException e) {
-        // ignore and look in superclass
-        if (recursive) {
-          clazz = clazz.getSuperclass();
-        } else {
-          break;
-        }
-      }
+  /**
+   * Gets a static field value of a given class
+   * @param targetClass the class that holds the requested static field.
+   * @param fieldName the name of the field
+   * @param recursive flag to lookup for the field in subclasses or not
+   * @param <T> the type of the field
+   * @return the value of the field
+   * @throws NoSuchFieldException when the field is not found
+   * @throws IllegalAccessException when the field is not reachable.
+   */
+  public static <T> T getStaticFieldValue(Class<?> targetClass, String fieldName, boolean recursive)
+      throws NoSuchFieldException, IllegalAccessException {
+    Field field = getField(targetClass, fieldName, recursive);
+    if (!Modifier.isStatic(field.getModifiers())) {
+      throw new IllegalAccessException(String.format("The %s field of %s class is not static", fieldName, targetClass.getName()));
     }
+    field.setAccessible(true);
```

Comment:
set accessible to its original value after getting the value of the field, in a finally block